src/interchange/avro.rs

```diff
@@ -929,12 +929,27 @@ impl Encoder {
                     ScalarType::TimestampTz => {
                         Value::Timestamp(datum.unwrap_timestamptz().naive_utc())
                     }
+                    // Duration Avro format: https://avro.apache.org/docs/current/spec.html#Duration
+                    // Compress values down into months, days, microseconds to ensure
+                    // values fit.
                     ScalarType::Interval => Value::Fixed(12, {
                         let iv = datum.unwrap_interval();
                         let mut buf = Vec::with_capacity(12);
-                        buf.extend(&(iv.months as i32).to_le_bytes());
-                        buf.extend(&0i32.to_le_bytes());
-                        buf.extend(&(iv.duration.as_millis() as i32).to_le_bytes());
+                        // Avro doesn't support negative durations.
+                        let mut dur = iv.duration.abs();
+                        // We assume 24 hours per day w/o any complexity because
+                        // there aren't more details.
+                        let ns = dur % (24 * 60 * 60 * 1_000_000_000);
+                        dur /= 24 * 60 * 60 * 1_000_000_000;
+                        // We assume 30 days per month like in Postgres because
+                        // there aren't more details about the Avro
+                        // implementation.
+                        let days = dur % 30;
+                        dur /= 30;
+                        let months = iv.months.abs() + dur as i64;
+                        buf.extend(&(months as u32).to_le_bytes());
+                        buf.extend(&(days as u32).to_le_bytes());
+                        buf.extend(&((ns / 1_000_000) as u32).to_le_bytes());
```

Comment:
I think this was more correct in one dimension before, although the overflow was definitely a problem. Specifically the Avro spec has this to say:

> The duration logical type represents an amount of time defined by a number of months, days and milliseconds. This is not equivalent to a number of milliseconds, because, depending on the moment in time from which the duration is measured, the number of days in the month and number of milliseconds in a day may differ.

In other words the months, days, and seconds fields are all actually different dimensions of the interval, much like we separate out months and nanoseconds. Converting nanoseconds to days/months will change the meaning of the interval, by the letter of the spec.

That said... it turns out the Java Avro client doesn't even support (!!!!) this type. https://issues.apache.org/jira/browse/AVRO-2123. So probably we should just ditch using this format and make up our own extremely convenient format instead. Either the string representation or just blatting down the two ints, probably.

But given that almost certainly no one is using this, doesn't need to block this PR!

Reply:
YOLO'ed our own format in here.